### PR TITLE
Show a GitButler indicator in Source Control for GitButler-managed repos

### PR DESCRIPTION
**Pull request by brennanb2025**
## What changes

When a worktree is checked out on GitButler's workspace branch (`gitbutler/workspace`, or the legacy `gitbutler/integration`) **and** the repository has GitButler metadata in its common git dir, the Source Control panel now shows a small informational **GitButler** badge. Its tooltip explains the practical consequence: *Orca's Source Control acts on the `gitbutler/workspace` branch, not your virtual branches.*

## What does NOT change (important boundary)

This is **awareness only**, not integration. Orca keeps using plain git. The badge:
- does **not** read, render, or write GitButler virtual branches / stacks,
- does **not** delegate any Orca git action to GitButler,
- does **not** block, disable, or gate any Source Control action.

It exists to prevent the most likely footgun — committing through Orca while expecting GitButler virtual-branch semantics — and to give Orca a detected signal to build on.

This PR is the small, safe **foundation slice** of the linked feature request. The two headline options in that request (full GitButler integration, or a generic source-control extension/provider API) are both genuinely XL one-way-door designs and are **deferred** — I've posted a concrete proposal as a comment on the issue and flagged that they need a product scoping decision before any code.

## Design approach

- Detection is filesystem-only and lives in a small focused module (`src/main/git/gitbutler-detection.ts`). It requires **both** the workspace branch name (after stripping `refs/heads/`) **and** a `gitbutler` *directory* in the repo's **common** git dir. The AND is load-bearing: branch-name-only would false-positive on a coincidentally named branch; metadata-only would false-positive on stale metadata while a normal branch is checked out.
- For linked worktrees, `resolveGitDir` returns the per-worktree dir, so detection derives the **common** dir from git's `commondir` file (GitButler stores metadata in the common dir). Since Orca is worktree-centric, getting this right is the common case, not a corner case.
- The filesystem probe is **gated behind the branch-name check**, so normal repos do zero extra fs work on the git-status path.
- The signal rides a new optional `GitStatusResult.managedBy?: 'gitbutler'` (string enum so a future manager can extend it; optional + absent-by-default keeps consumers and the wire protocol backward compatible), flows through the existing `setGitStatus` reducer into a `gitStatusManagedByByWorktree` map, and renders via a `<GitButlerManagedBadge>` modeled on the existing `DetachedHeadBadge`.

## Headline behavior status

**Implemented end-to-end in production code** (local provider). Verified in the running app: the badge appears on a `gitbutler/workspace` worktree with metadata and disappears when a normal branch is checked out — even with `.git/gitbutler` still present (the AND guard working live). The `runGetStatus` path emitting `managedBy` is covered by an integration test, not just the helper in isolation.

**Accepted boundary:** detection runs on the **local provider only**. On SSH/remote, `managedBy` is simply absent → no badge (graceful, not wrong — GitButler is a desktop-local tool). Remote parity would need a remote `.git/gitbutler` probe; noted as follow-up, not built here.

## Validation

- **Design review:** ran an automated design-review/fix loop. It caught a real correctness bug (linked-worktree detection would have silently never fired because it probed the per-worktree dir instead of the common dir) — fixed before implementation.
- **Completeness verification:** confirmed every design requirement, edge case, and validation item is present; closed the one gap (added the `runGetStatus` integration test proving the production path).
- **Perf audit:** performance-neutral. No added git subprocess; zero fs work for non-GitButler repos (branch-name gate before any fs call); the new store map mirrors the existing `gitStatusHeadByWorktree` change-detection so an unchanged status doesn't perturb store identity or re-render. Added a test asserting the gate short-circuits before any git-dir resolution.
- **Code-review loop:** two independent reviewers per round. Round 1 surfaced a P2 (the metadata check must require a *directory*, not accept a regular file named `gitbutler`) — fixed with an `isDirectory` check plus file-vs-dir and absolute-`commondir` tests. A flagged P1 (a common-dir resolver "already exists") was assessed and deliberately not consolidated: the two prior copies aren't clean drop-ins (one takes a `Repo` and uses a path heuristic rather than reading the authoritative `commondir`; the other is a private inner block of a GitHub-config function), so consolidating would reach into unrelated subsystems for no behavioral gain. Final round: both reviewers clean.
- **Merge-confidence (real app):** validated end-to-end in Electron with screenshots of badge-present, tooltip, and badge-absent states (attached to this PR conversation).
- **Static checks + tests:** `pnpm run typecheck` clean; oxlint clean on changed files; 93 targeted tests pass (`gitbutler-detection`, `status`, `editor` reducer).

## Files

- `src/main/git/gitbutler-detection.ts` (+ test) — detection helper
- `src/main/git/status.ts` (+ test) — wires `managedBy` into `runGetStatus`
- `src/shared/git-status-types.ts` — `managedBy?: 'gitbutler'` on `GitStatusResult`
- `src/renderer/src/store/slices/editor.ts` (+ test) — `gitStatusManagedByByWorktree` map + reducer
- `src/renderer/src/components/GitButlerManagedBadge.tsx` — the badge
- `src/renderer/src/components/right-sidebar/SourceControl.tsx` — selector + conditional render

## Remaining risks / non-actionable notes

- SSH/remote shows no badge (accepted, documented above).
- Three independent common-git-dir resolvers now exist in the tree; consolidating them is a reasonable follow-up but out of scope here.

Made with [Orca](https://github.com/stablyai/orca) 🐋

### CODE REVIEW

**Comment by brennanb2025** [comment]
### Manual validation evidence (Electron, real app)

Validated end-to-end in a dev build of this branch. Three full-window screenshots were captured (uncropped, with surrounding app context). Summary of what each shows:

1. **Badge present** — repo on `gitbutler/workspace` with a `.git/gitbutler/` directory. The **GitButler** badge (Layers icon, muted/informational styling) renders in the Source Control header row, directly under the `vs main` base-ref line. Backing state `gitStatusManagedByByWorktree[activeWorktreeId] === 'gitbutler'` was populated by the real `runGetStatus → setGitStatus` flow (never set by hand).
2. **Tooltip** — hovering the badge shows the exact copy: *"This repo is managed by GitButler. Orca's Source Control acts on the gitbutler/workspace branch, not your virtual branches."*
3. **Badge absent (false-positive guard, live)** — after `git checkout -b feature/x` in the same repo **with `.git/gitbutler/` still present**, the next status poll cleared the key and the badge row collapsed. This is the case the branch-name AND metadata guard exists for, confirmed in the running app.

No feature-related console errors. SSH/remote was reasoned through (badge correctly absent on remote — detection is local-provider-only, no regression) rather than run live; cross-platform path handling uses `path.join`/`path.resolve` throughout with a unit test for both relative and absolute `commondir` forms.

_(Screenshot image files were kept out of the repo per the project's evidence policy; the verified states are described above. Happy to drop the PNGs into this conversation if useful.)_

File: src/renderer/src/components/GitButlerManagedBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            className
+          )}
+        >
+          <Layers className="size-2.5" />
+          GitButler
+        </Badge>
+      </TooltipTrigger>
+      <TooltipContent side={side} sideOffset={8}>
+        This repo is managed by GitButler. Orca&apos;s Source Control acts on the
+        gitbutler/workspace branch, not your virtual branches.
```
[the remark this answers is not in the document]
Comment (brennanb2025) [reply src/renderer/src/components/GitButlerManagedBadge.tsx]: Good catch — fixed in 47035f0a6. Reworded so it no longer hardcodes `gitbutler/workspace`, since detection also covers the legacy `gitbutler/integration` branch.
